hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/SnapshotLoadQuerySplitter.java

```diff
@@ -118,6 +124,32 @@ public QueryInfo getNextCheckpoint(Dataset<Row> df, QueryInfo queryInfo, Option<
         .orElse(queryInfo);
   }
 
+  public Option<CheckpointWithPredicates> getNextCheckpoint(Dataset<Row> df, QueryContext queryContext,
+                                                            Option<SourceProfileSupplier> sourceProfileSupplier) {
+    // the start instant would be included into the final query result. So we need to get
+    // a strictly lower timestamp to have query splitter include the start instant
+    Option<CheckpointWithPredicates> nextCheckpointWithPredicates =
+        getNextCheckpointWithPredicates(df, instantTimeMinusMillis(queryContext.getBeginInstant().get(), 1));
```

Comment:
Can we define the behavior through range type?

Reply:
The `getNextCheckpointWithPredicates` API in `SnapshotLoadQuerySplitter` is changed to take `QueryContext` and return completion time as the checkpoint directly, so all such logic of conversion between instant time and completion is no longer needed.